pkgs/applications/video/screenkey/default.nix

```diff
@@ -1,50 +1,60 @@
 { lib
 , substituteAll
-, buildPythonApplication
-, fetchFromGitHub
-, distutils_extra
-, setuptools-git
+, fetchFromGitLab
+# native
 , intltool
-, pygtk
-, libX11
-, libXtst
 , wrapGAppsHook
-, gnome3
+, file
+# not native
+, xorg
+, gobject-introspection
+, gtk3
+, python3
 }:
-buildPythonApplication rec {
+
+python3.pkgs.buildPythonApplication rec {
   pname = "screenkey";
-  version = "0.9";
+  version = "1.2";
 
-  src = fetchFromGitHub {
-    owner = "wavexx";
+  src = fetchFromGitLab {
+    owner = "screenkey";
     repo = "screenkey";
-    rev = "screenkey-${version}";
-    sha256 = "14g7fiv9n7m03djwz1pp5034pffi87ssvss9bc1q8vq0ksn23vrw";
+    rev = "v${version}";
+    sha256 = "1x13n57iy2pg3h3r994q3g5nbmh2gwk3qidmmcv0g7qa89n2gwbj";
   };
 
-  patches = [
-    (substituteAll {
-      src = ./paths.patch;
-      inherit libX11 libXtst;
-    })
-  ];
-
   nativeBuildInputs = [
-    distutils_extra
-    setuptools-git
+    # Shouldn't be needed once https://gitlab.com/screenkey/screenkey/-/issues/122 is fixed.
     intltool
-
+    file
     wrapGAppsHook
   ];
+  # Fix CDLL python calls for non absolute paths of xorg libraries
+  preBuild = ''
+    substituteInPlace Screenkey/xlib.py \
+      --replace libX11.so.6 ${lib.getLib xorg.libX11}/lib/libX11.so.6 \
+      --replace libXtst.so.6 ${lib.getLib xorg.libXtst}/lib/libXtst.so.6
+  '';
 
   buildInputs = [
-    gnome3.adwaita-icon-theme
+    xorg.libX11
+    xorg.libXtst
```

Comment:
Are these used for anything but the patch?

Reply:
IIRC we handle this by splicing and the hooks would not use spliced derivation in either case – Nixpkgs has no idea we are doing something with derivations in the hooks like it would in inputs and Nix does not care.

Reply:
Putting these xorg deps in `buildInputs` is more explicit IMO. But I'm not strongly opinionated so whatever you say @jtojnar.

Reply:
Adding a dependency to `buildInputs` when they are not used by anything is superfluous and, in my opinion, confusing. `buildInputs` are already too overloaded for my liking (even by things like `autoPatchelfHook`), no need to add documentation as its use case.

Reply:
Fair enough.